src/app/url-parser.service.ts

```diff
@@ -50,23 +51,33 @@ export type CovidGraphDefinition = {
   // Add more as needed
 };
 
+const urlToGraphs = (): CovidGraphDefinition[] =>
+  window.location.hash
+    .slice(1)
+    .split('#')
+    .map(parseQueryString)
+    .map(userDefinedGraphProperties => ({
+      ...DEFAULT_GRAPH_PROPERTIES,
+      ...userDefinedGraphProperties,
+      locations: replaceDashWithCommaSpace(
+        userDefinedGraphProperties.locations || []
+      ),
+    }));
+
 @Injectable({
   providedIn: 'root',
 })
 export class UrlParserService {
```

Comment:
Actually can you revert this last commit for now (and then you can just merge this PR straight away) and re-open as a separate PR with just this commit? I have some ideas of how to improve this a bit, but that'll be slightly more involved